src/account/service.ts

```diff
@@ -0,0 +1,34 @@
+import { Md5 } from "ts-md5";
+import createError from "http-errors";
+import { MailChimpClient } from "../mailchimp";
+import type { UpdateTagsRequest } from "./models";
+
+const updateTags = async (requestBody: UpdateTagsRequest): Promise<void> => {
+  const tags = (requestBody.addTags ?? [])
+    .map((tag): { name: string; status: "active" | "inactive" } => ({
+      name: tag,
+      status: "active",
+    }))
+    .concat(
+      (requestBody.removeTags ?? []).map(
+        (tag): { name: string; status: "active" | "inactive" } => ({
+          name: tag,
+          status: "inactive",
+        })
+      )
+    );
+
+  const response = await MailChimpClient.lists.updateListMemberTags(
+    process.env["MAILCHIMP_COMMUNITY_LIST_ID"] ?? "",
+    Md5.hashStr(requestBody.emailFromAuthToken),
+    { tags }
+  );
```

Comment:
I think this should be in a try/catch or otherwise handle a MailChimp failure a little more intentionally. My email wasn't present on Mailchimp (because it's a local test email, I suppose, and we don't generally add those to Mailchimp) and so I got a 404 back. If something goes wrong here then I think it would be good to catch that and send a JSON error. Currently we just pass on the "not found" HTML.

Reply:
Added middleware to convert these and any other third party errors with `status` or `statusCode` fields to JSON